### ISSUE TITLE
Complete the `make relay` command

### ISSUE THREAD

**Issue by cmwaters** [celestiaorg/celestia-zkevm-ibc-demo#31]
To transfer tokens from simapp to the reth rollup requires the following steps:

- [x] Submit `MsgTransfer` to simapp commencing the process. See https://github.com/celestiaorg/celestia-zkevm-ibc-demo/pull/50?notification_referrer_id=NT_kwDOADhxZ7MxMzkxOTc1NDc4MDozNjk5MDQ3
- [x] Update the tendermint client on the reth rollup (querying the prover process for the groth16 proof proving the simapps state transition from a previous trusted height to a new one) https://github.com/celestiaorg/celestia-zkevm-ibc-demo/pull/67
- [x] Query the membership proof of the commitment on simapp and attach that proof to the transfer packet that needs to be sent to the reth rollup (I think to the router smart contract) See https://github.com/celestiaorg/celestia-zkevm-ibc-demo/pull/186

We can omit acknowledgments back since they're not required for the EVM to update balances.

To transfer tokens from the reth rollup to simapp requires the following steps:

- [x] #220
- [x] #221
- [ ] #222
- [ ] #247
- [ ] #245

Similarly, we can omit acknowledgments back since they're not required for the simapp to update balances.

**Comment by rootulp** [comment]
AFAIK this is blocked on https://github.com/celestiaorg/celestia-zkevm-ibc-demo/issues/30 because the clients and channel need to be set up before we try to update the clients and use the connection.

I can explore creating a mock client and channel if https://github.com/celestiaorg/celestia-zkevm-ibc-demo/issues/30 isn't ready yet cc: @ninabarbakadze

**Comment by cmwaters** [comment]
Yes you won't be able to test this flow until #30 is completed but I think you can already write most of the code

**Comment by cmwaters** [comment]
I think you can get a lot of inspiration from: https://github.com/cosmos/solidity-ibc-eureka/blob/main/e2e/interchaintestv8/ibc_eureka_test.go

**Comment by rootulp** [comment]
>  Submit MsgTransfer to simapp commencing the process

Nina already wrote logic for broadcasting + confirming transactions inside https://github.com/celestiaorg/celestia-zkevm-ibc-demo/pull/34. May need to generalize it.

>  Update the tendermint client on the reth rollup (querying the prover process for the groth16 proof proving the simapps state transition from a previous trusted height to a new one)

1. Query the prover GRPC service that Callum just wrote. It is running on port ___. Need a README or demo on how that works.
2. Submit a tx on the Reth roll-up to the tendermint light client with the proof. Invoke `sp1ics07tendermint`'s `NewContract` so that we can call... (needs more research).

**Comment by rootulp** [comment]
For Tendermint chains:
- Multiple messages (UpdateClient, MsgTransfer) get bundled into one tx.

For EVM chains:
- We don't know which order we need to send the txs in.

**Comment by rootulp** [comment]
Clarification: steps 2 - 5 are responsibilities of the relayer. See https://github.com/celestiaorg/celestia-zkevm-ibc-demo/pull/63 for a more detailed breakdown of what each step entails.

**Comment by rootulp** [comment]
A few notes on the relayer from sprint planning:
- Don’t create a separate service or process for the relayer. 
    - Do something similar to the ibc-go repo e2e tests which uses Go code to execute the steps of what a relayer would do.
    - IBC team built a one-off relayer inside the solidity IBC repo. Written in Rust. Just exists for end-to-end tests. Would have to be modified for our Groth16 proofs. Use for inspiration.
    - Look at Voyager (built by Union) for inspiration but it uses a different EVM contract.
- The solidity implementation of IBC Eureka has event emission.

**Comment by rootulp** [comment]
https://github.com/celestiaorg/celestia-zkevm-ibc-demo/pull/67

**Comment by rootulp** [comment]
A more detailed write-up exists in the [README](https://github.com/celestiaorg/celestia-zkevm-ibc-demo/blob/16e626d0c9d1260adf4cc72aa06f61818f3de552/README.md?plain=1#L83-L89).

**Comment by rootulp** [comment]
Renamed the issue b/c `make transfer` was completed by step 1 of this issue. The remainder is now `make relay`. We can combine those commands at the end of this issue if people desire.

**Comment by ninabarbakadze** [comment]
Working on the last two steps

**Comment by rootulp** [comment]
Closing this b/c it's basically done. We just need to merge these two PRs:
1. https://github.com/celestiaorg/celestia-zkevm-ibc-demo/pull/257
2. https://github.com/celestiaorg/celestia-zkevm-ibc-demo/pull/244
